packages/synapse-react-client/src/utils/hooks/useUploadFileEntity/useTrackFileUploads.ts

```diff
@@ -155,6 +162,19 @@ export function useTrackFileUploads() {
     })
   }
 
+  // The total number of bytes that are being uploaded, but do not yet count against the project storage limits.
+  // Used to prevent the uploader from exceeding storage limits before uploading the file (the backend won't provide an error response until after the file is uploaded)
+  const bytesPendingUpload = [...trackedUploadProgress].reduce(
+    (acc, [file, progress]) => {
+      if (UNTRACKED_BYTE_UPLOAD_STATES.includes(progress.status)) {
+        return acc + file.size
+      }
+
+      return acc
+    },
+    0,
+  )
```

Comment:
This sequence is possible:

1. You have a project with a 100GB limit, 0 files uploaded
2. You open the uploader and add a 90 GB file
3. Before (or very shortly after) that file finishes uploading, you try to add another 50GB file (this was not possible in the SWC uploader unless you opened two windows)

WDYT about this idea? Track the size of all files that have been uploaded or will be uploaded in the current set of files. We will add that to the reported `sumFileBytes` to decide whether to block uploads. The main risk is that files could be double-counted by both the server and the client, though I set the `staleTime` to infinity on the usage query to attempt to prevent that.

Reply:
I think the logic that you have implemented is ok.  If files are double-counted, what is the UX?

Reply:
The UI will erroneously block you from uploading any new files that push you over the value `limit - size of double counted files`. If we erroneously blocked the upload, you would be able to start the upload if you reloaded the page or opened a new window.